src/test/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicatorTest.java

```diff
@@ -0,0 +1,156 @@
+
+	@BeforeEach()
+	public void setUp() throws Exception {
+		haivisionX4DecoderCommunicator.setHost("mx4d.demo.haivision.com");
+		haivisionX4DecoderCommunicator.setPort(443);
+		haivisionX4DecoderCommunicator.setLogin("operator");
+		haivisionX4DecoderCommunicator.setPassword("supervisor");
+		haivisionX4DecoderCommunicator.init();
+		haivisionX4DecoderCommunicator.connect();
+	}
+
+	@AfterEach()
+	public void destroy() throws Exception {
+		haivisionX4DecoderCommunicator.disconnect();
+	}
+
+	/**
+	 * Test HaivisionX4DecoderCommunicator.getMultipleStatistics successful with valid username password
+	 * Expected retrieve valid device monitoring data
+	 */
+	@Tag("RealDevice")
+	@Test
+	void testHaivisionX4DecoderCommunicatorGetMonitoringDataSuccessful() throws Exception {
+		ExtendedStatistics extendedStatistics = (ExtendedStatistics) haivisionX4DecoderCommunicator.getMultipleStatistics().get(0);
+		Map<String, String> stats = extendedStatistics.getStatistics();
+
+		String deviceInfoGroup = MonitoringMetricGroup.DEVICE_INFO.getName() + DecoderConstant.SPACE + DecoderConstant.HASH;
+		String decoderStatisticGroup = MonitoringMetricGroup.DECODER_STATISTICS.getName() + DecoderConstant.SPACE + 1 + DecoderConstant.HASH;
+
+		Assertions.assertEquals(MonitoringData.SERIAL_NUMBER.getData(), stats.get(deviceInfoGroup + DeviceInfoMetric.SERIAL_NUMBER.getName()));
+		Assertions.assertEquals(MonitoringData.HARDWARE_REVISION.getData(), stats.get(deviceInfoGroup + DeviceInfoMetric.HARDWARE_REVISION.getName()));
+		Assertions.assertEquals(MonitoringData.PART_NUMBER.getData(), stats.get(deviceInfoGroup + DeviceInfoMetric.PART_NUMBER.getName()));
+		Assertions.assertEquals(MonitoringData.FIRMWARE_VERSION.getData(), stats.get(deviceInfoGroup + DeviceInfoMetric.FIRMWARE_VERSION.getName()));
+		Assertions.assertEquals(MonitoringData.DECODER_ID.getData(), stats.get(decoderStatisticGroup + DecoderMonitoringMetric.DECODER_ID.getName()));
+	}
+
+	/**
+	 * Test HaivisionX4DecoderCommunicator.getMultipleStatistics will throw ResourceNotReachableException when Username or Password are incorrect
+	 * Expected throw ResourceNotReachableException "Username and Password are incorrect";
+	 */
+	@Tag("RealDevice")
+	@Test
+	void testHaivisionX4DecoderCommunicatorLogInFailed() {
+		HaivisionX4DecoderCommunicator haivisionX4DecoderCommunicatorSpy = Mockito.spy(HaivisionX4DecoderCommunicator.class);
+		haivisionX4DecoderCommunicatorSpy.setHost("mx4d.demo.haivision.com");
+		Mockito.when(haivisionX4DecoderCommunicatorSpy.getLogin()).thenReturn("123");
+		Mockito.when(haivisionX4DecoderCommunicatorSpy.getPassword()).thenReturn("456");
+		ResourceNotReachableException exception = Assertions.assertThrows(ResourceNotReachableException.class, () -> {
+			ExtendedStatistics extendedStatistics = (ExtendedStatistics) haivisionX4DecoderCommunicatorSpy.getMultipleStatistics().get(0);
```

Comment:
extendedStatistics unused assign, please remove it

Reply:
please check the other as well

Reply:
I fixed it